### ISSUE TITLE
Build Issues and Solutions for wardenprotocol

### ISSUE THREAD

**Issue by jemi2k** [warden-protocol/wardenprotocol#1157]
Description:
I encountered several issues while building the wardenprotocol project. Below are the details of the issues and the solutions applied:

Issue 1: Undefined `secp256k1.RecoverPubkey and secp256k1.VerifySignature`

Error Message:
```
undefined: secp256k1.RecoverPubkey
undefined: secp256k1.VerifySignature
```
Solution: Replaced `secp256k1` functions with `ethcrypto` functions from `github.com/ethereum/go-ethereum/crypto`

Issue 2: Type Mismatch in `legacy.go`

Error Message:

`cannot use runtime.NewKVStoreService(app.GetKey(wasmtypes.StoreKey)) (value of type store.KVStoreService) as "cosmossdk.io/store/types".StoreKey value in argument to wasmkeeper.NewKeeper: store.KVStoreService does not implement "cosmossdk.io/store/types".StoreKey (missing method Name)`

Solution: Passed `app.GetKey(wasmtypes.StoreKey)` directly to `wasmkeeper.NewKeeper`

Issue 3: `AllCapabilities()` Return Type Mismatch

Error Message:

cannot use `AllCapabilities() (value of type []string)` as string value in argument to `wasmkeeper.NewKeeper`

Solution: Joined the capabilities slice into a single string using `strings.Join(AllCapabilities(), ",")`.

Issue 4: Missing strings Package Import

Error Message:

undefined: `strings`

Solution: Imported the strings package.

**Comment by pitasi** [comment]
Hi, can you share exactly what were the steps that you did and led to those errors?

I can do this:
```
git clone git@github.com:warden-protocol/wardenprotocol
cd wardenprotocol
just localnet
```

and works fine on my machine. But I'd like to better understand what exactly didn't work for you.

**Comment by jemi2k** [comment]
I don't remember specifically at the moment but I was trying to deploy an EVM and WASM contract on Chiado. then, I also tried deploying a cross-chain smart contract application using gmp. I encountered some errors, as listed above. some of the errors were typos, but if any critical issues arise and cause errors, I will create a PR

**Comment by pitasi** [comment]
Absolutely, I'm closing this for now, but feel free to open new issues in the future with steps to reproduce so we can provide a better help! cheers
